cluster/gce/windows/k8s-node-setup.psm1

```diff
@@ -1113,13 +1113,33 @@ $STACKDRIVER_ROOT = 'C:\Program Files (x86)\Stackdriver'
 # sometimes is unstoppable, so we work around it by killing the processes.
 function Restart-StackdriverLoggingAgent {
   Stop-Service -NoWait -ErrorAction Ignore StackdriverLogging
-  # TODO: check periodically to lower the wait time
-  Start-Sleep 10
+ 
+  # Wait (if necessary) for service to stop.
+  $timeout = 10
+  $stopped = (Get-service StackdriverLogging).Status -eq 'Stopped'
+  for ($i = 0; $i -lt $timeout -and !($stopped); $i++) {
+      Start-Sleep 1
+      $stopped = (Get-service StackdriverLogging).Status -eq 'Stopped'
+  }
+
   if ((Get-service StackdriverLogging).Status -ne 'Stopped') {
     # Force kill the processes.
     Stop-Process -Force -PassThru -Id (Get-WmiObject win32_process |
       Where CommandLine -Like '*Stackdriver/logging*').ProcessId
   }
+
+  # Wait until process has stopped.
```

Comment:
We only need this code to run after we've tried to force-kill the process, right? If so then it would be slightly preferable to put it inside the force-kill block above.